dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaClusterCPUChartTask.java

```diff
@@ -0,0 +1,157 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.ClouderaManagerHandle.ClouderaClusterDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.hc.core5.net.URIBuilder;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The task collects hosts from Cloudera Manager <a
+ * href="https://cldr2-aw-dl-gateway.cldr2-cd.svye-dcxb.a5.cloudera.site/static/apidocs/resource_TimeSeriesResource.html">TimeSeries
+ * API</a> The chart is for whole cluster is available on {@code /cmf/home/} and {@code
+ * cmf/clusters/{clusterId}/status} pages in Cloudera Manager UI. <b/> The query to chart is written
+ * on <a
+ * href="https://docs.cloudera.com/documentation/enterprise/latest/topics/cm_dg_tsquery.html">tsquery</a>
+ * language.
+ */
+public class ClouderaClusterCPUChartTask extends AbstractClouderaManagerTask {
+  enum TimeSeriesAggregation {
+    RAW,
+    TEN_MINUTELY,
+    HOURLY,
+    SIX_HOURLY,
+    DAILY,
+    WEEKLY,
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(ClouderaCMFHostsTask.class);
+  /*
+    SELECT cpu_percent_across_hosts WHERE entityName = "1546336862" AND category = CLUSTER
+  */
+  private static final String TS_CPU_QUERY_TEMPLATE =
+      "SELECT cpu_percent_across_hosts WHERE entityName = \"%s\" AND category = CLUSTER";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final ZonedDateTime from;
+  private final TimeSeriesAggregation tsAggregation;
+
+  public ClouderaClusterCPUChartTask() {
+    this(Instant.now().atZone(ZoneId.of("UTC")), TimeSeriesAggregation.RAW);
+  }
+
+  public ClouderaClusterCPUChartTask(ZonedDateTime from, TimeSeriesAggregation tsAggregation) {
+    super(
+        String.format(
+            "cmf-cluster-cpu-%s-%s.jsonl",
+            from.toString().replaceAll("[-:\\[\\]ZUTC.]", "").toLowerCase(),
+            tsAggregation.toString().toLowerCase()));
+    this.from = from;
```

Comment:
I understand that it's name of parameter in Cloudera API, but let's use some better name in our code.
It might not so obvious for  other team members.

Reply:
Moreover, I think we don't need this filed.
What we need it's numberOfDays `int` and aggregation `enum`, all the calculation with date and timezone should be encapsulated.

Reply:
Fixed.